### DISCUSSION TITLE
Have more details on "Error in binding" error messages

### DISCUSSION THREAD

**Discussion by Dragorn421** [AvaloniaUI/Avalonia#16059] [Q&A] [answered]
Hi,

I have a somewhat large application in which occasionally I get this kind of message in the debug output:

> [Binding]Error in binding to 'Avalonia.Controls.Primitives.SelectingItemsControl+BindingHelper'.'Value': 'Null value in expression '{empty}' at ''.'(BindingHelper #22942434)

With a debugger attached breaking on all exceptions, no breaking happens, so it doesn't seem this is linked to an exception. And not breaking means I can't get more context on the error.

How do I get more information/context on which window/control is relevant to such an error?
For example, is there a way to make the output more verbose?

**Comment by maxkatz6** [comment]
There are multiple issues on that topic already.
But generally, please try 11.1 builds, it should be way less noisy with false positives.

**Reply by Dragorn421** [reply]
I have not found any discussion or issue about "Error in binding" messages, could you point me to one with existing answers if there's one? I don't just want the warning gone, I'd like to know how to debug it

The warning does not appear using Avalonia 11.1.0-rc1

**Reply by stevemonaco** [reply, marked as answer]
It was an issue with the old binding implementation, so there's no reason for app developers to debug. It requires digging into the source code.

`BindingHelper` isn't part of control trees, so you can't crawl/search the visual tree for a control with hash of: `#22942434`. See https://github.com/AvaloniaUI/Avalonia/issues/15633 and the Debugging demo I have linked there for approaches that work when the control is part of the visual tree, like usual. I haven't had time to implement this functionality into DevTools, but these logging / lookup approaches should solve nearly all app developer issues regarding tracking down which control is experiencing a binding error. At least ones you might be able to fix.

**Comment by stevemonaco** [comment]
FYI, you can customize logging through [`LogEventLevel`](https://docs.avaloniaui.net/docs/guides/implementation-guides/logging-errors-and-warnings).

It's probably a false positive (especially if you're using a release prior to 11.1), but `BindingHelper` is related to item selection bindings (ie. `SelectedItem`).

**Reply by Dragorn421** [reply]
Logging all levels (Trace+) didn't provide any more information.

I'm using Avalonia 11.0.10, I tried Avalonia 11.1.0-rc1 and the warning does disappear then.
